java/ql/src/semmle/code/java/security/XPath.qll

```diff
@@ -0,0 +1,65 @@
+/** Provides classes to reason about XPath vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+
+/**
+ * An abstract type representing a call to interpret XPath expressions.
+ */
+class XPathSink extends MethodAccess {
+  /**
+   * Gets the argument representing the XPath expressions to be evaluated.
+   */
+  abstract Expr getSink();
+}
+
+/** The class `javax.xml.xpath.XPath` */
+class XPath extends RefType {
+  XPath() { this.hasQualifiedName("javax.xml.xpath", "XPath") }
+}
+
+/** A call to `XPath.evaluate` or `XPath.compile` */
+class XPathEvaluateOrCompile extends XPathSink {
+  XPathEvaluateOrCompile() {
+    exists(Method m | this.getMethod() = m and m.getDeclaringType() instanceof XPath |
+      m.hasName(["evaluate", "compile"])
```

Comment:
Maybe it would be good to consider [`evaluateExpression`](https://docs.oracle.com/en/java/javase/16/docs/api/java.xml/javax/xml/xpath/XPath.html#evaluateExpression(java.lang.String,java.lang.Object)) as well?
(Would require changing the QLDoc of this class then as well)

Reply:
I changed the class a bit to include that case, thanks! See ab543de